app/src/backend/formulas.ts

```diff
@@ -61,6 +72,189 @@ export function handleDirectMeasureFormula(
+
+    const AmountOfWasteForWasteTypeI = massOfIncineratedWaste * WasteFractionI;
+
+    const CH4EmissionFactorForWasteTypeI =
+      CH4EmissionFactor[technology]?.[boilerType];
+
+    const NO2EmissionFactorForWasteTypeI =
+      NO2EmissionFactor[wasteType]?.[technology];
+
+    if (CH4EmissionFactorForWasteTypeI == null) {
+      throw new createHttpError.BadRequest(
+        `No CH4 emission factor found for ${technology}, ${boilerType}`,
+      );
+    }
+
+    if (NO2EmissionFactorForWasteTypeI == null) {
+      throw new createHttpError.BadRequest(
+        `No NO2 emission factor found for ${wasteType}, ${technology}`,
+      );
+    }
+
+    (totalCH4Emission +=
+      AmountOfWasteForWasteTypeI * CH4EmissionFactorForWasteTypeI * 10 ** -3),
+      (totalN2OEmissions +=
+        AmountOfWasteForWasteTypeI * NO2EmissionFactorForWasteTypeI * 10 ** -3);
+
+    // calculate CO2 emissions
+
+    const formulaInputs = await db.models.FormulaInput.findAll({
+      where: {
+        [`metadata.waste-type`]: formulaInputsMapping[wasteType] as string,
+        gas: "CO2",
+        formulaName: "incineration-waste",
+        gpcRefno: inventoryValue.gpcReferenceNumber,
+        region: "world",
+      },
+    });
+
+    const dryMatterInput = formulaInputs.find(
+      (input) => input.parameterCode === "dmi",
+    )?.formulaInputValue;
+
+    const fractionOfCarbonInput = formulaInputs.find(
+      (input) => input.parameterCode === "CFi",
+    )?.formulaInputValue;
+
+    const fractionOfFossilCarbonInput = formulaInputs.find(
+      (input) => input.parameterCode === "FCFi",
+    )?.formulaInputValue;
+
+    const fractionOfFossilCarbonI = fractionOfFossilCarbonInput ?? 1;
+
+    const dryMatterContentI = dryMatterInput ?? 1;
+
+    const fractionOfCarbonI = fractionOfCarbonInput ?? 1;
+
+    if (!dryMatterInput) {
+      console.warn(
+        `dryMatterContentI is missing for ${wasteType} a default of 1 used`,
+      );
+    }
+
+    if (!fractionOfCarbonInput) {
+      console.warn(
+        `fractionOfCarbonI is missing for ${wasteType} a default of 1 used`,
+      );
+    }
+
+    if (!fractionOfFossilCarbonInput) {
+      console.warn(
+        `fractionOfFossilCarbonI is missing for ${wasteType} a default of 1 used`,
+      );
+    }
+
+    const oxidationFactorI = CO2OxidationFactor[technology];
+
+    totalPartialCO2Emissions +=
+      WasteFractionI *
+      dryMatterContentI *
+      fractionOfCarbonI *
+      fractionOfFossilCarbonI *
+      oxidationFactorI;
+  }
+
+  const totalCO2Emissions =
+    massOfIncineratedWaste * totalPartialCO2Emissions * (44 / 12);
+
+  return [
+    {
+      gas: "CH4",
+      amount: BigInt(new Decimal(totalCH4Emission).trunc().toString()),
+    },
+    {
+      gas: "N2O",
+      amount: BigInt(new Decimal(totalN2OEmissions).trunc().toString()),
+    },
+    {
+      gas: "CO2",
+      amount: BigInt(new Decimal(totalCO2Emissions).trunc().toString()),
```

Comment:
If we are using DecimalJS now, we should return that instead of a truncated BigInt to keep the precision. Fine for now, we will do this once the other formulas are refactored to use Decimal.js as well :+1: